app_dart/lib/src/foundation/github_checks_util.dart

```diff
@@ -140,4 +140,55 @@ class GithubChecksUtil {
       output: output,
     );
   }
+
+  /// Retrieves all check runs that match the Git [ref].
+  ///
+  /// Optionally, you can supply [checkName], [status], and/or [filter] to refine the query.
+  Future<Stream<github.CheckRun>> listCheckRunsForRef(
+    Config cocoonConfig,
+    github.RepositorySlug slug, {
+    required String ref,
+    String? checkName,
+    github.CheckRunStatus? status,
+    github.CheckRunFilter? filter,
+  }) async {
+    const RetryOptions r = RetryOptions(
+      maxAttempts: 3,
+      delayFactor: Duration(seconds: 2),
+    );
+    return r.retry(
+      () async {
+        return _listCheckRunsForRef(
+          cocoonConfig,
+          slug,
+          ref: ref,
+          checkName: checkName,
+          status: status,
+          filter: filter,
+        );
+      },
+      retryIf: (Exception e) => true,
+      onRetry: (Exception e) => log.warning(
+        'listCheckRunsForRef fails for slug: ${slug.fullName}, ref: $ref, checkName: $checkName, status: $status, filter: $filter. Exception: ${e.toString()}',
+      ),
+    );
+  }
+
+  Future<Stream<github.CheckRun>> _listCheckRunsForRef(
+    Config cocoonConfig,
+    github.RepositorySlug slug, {
+    required String ref,
+    String? checkName,
+    github.CheckRunStatus? status,
+    github.CheckRunFilter? filter,
+  }) async {
+    final github.GitHub gitHubClient = await cocoonConfig.createGitHubClient(slug: slug);
+    return gitHubClient.checks.checkRuns.listCheckRunsForRef(
```

Comment:
Also just a heads up you might want to look at what this code is actually calling since this is basically a wrapper to a REST call. There might be ways you can get what you need by modifying the library. Just an fyi.

Reply:
Oh yea, I was glad to find this function since I was prototyping this implementation by looking at the API. https://docs.github.com/en/rest/checks/runs?apiVersion=2022-11-28#list-check-runs-for-a-git-reference

This is the command that I was using locally:
```bash
gh api -H "Accept: application/vnd.github+json" "/repos/flutter/plugins/commits/614d7c0df1eb03d2579118b4a0c3032dbedfa364/check-runs?check=Mac_x64%20ios_platform_tests_4_of_4%20master"
```